app/services/participant_profile_resolver.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class ParticipantProfileResolver
+  class << self
+    def call(participant_identity, course_identifier)
+      return unless participant_identity
+
+      if ParticipantProfile::ECT::COURSE_IDENTIFIERS.include?(course_identifier)
+        participant_identity
+          .participant_profiles
+          .active_record
+          .ects
+          .first
+      elsif ParticipantProfile::Mentor::COURSE_IDENTIFIERS.include?(course_identifier)
+        participant_identity
+          .participant_profiles
+          .active_record
+          .mentors
+          .first
+      elsif NPQCourse.identifiers.include?(course_identifier)
+        participant_identity
+          .participant_profiles
+          .joins(participant_identity: { npq_applications: :npq_course })
+          .where(npq_courses: { identifier: course_identifier })
+          .first
```

Comment:
there are some workarounds that have been implemented to handle bad data aka duplicates. this appears to be missing here which i foresee will add more problems to the pile if they are removed. For example there exists code elsewhere where the calling provider is passed in, this extra information can be used to determine the correct profile from their duplicates. without this we may end up attaching declarations to the incorrect profile?

Reply:
@asmega this class is mere copy past of what is already existing:
https://github.com/DFE-Digital/early-careers-framework/blob/develop/app/services/participants/early_career_teacher.rb#L11-L15

https://github.com/DFE-Digital/early-careers-framework/blob/develop/app/services/participants/mentor.rb#L11-L15

https://github.com/DFE-Digital/early-careers-framework/blob/develop/app/services/participants/npq.rb#L20-L31

I think this what has been causing us a bunch of pain. Hence I've collected those in a class, we will need to address those.

Reply:
See this PR for more detail

> this class is mere copy past of what is already existing

> https://github.com/DFE-Digital/early-careers-framework/blob/develop/app/services/participants/npq.rb#L20-L31

It's not though, as you see there is an additional `where` clause in the code you referenced but it's not present in the version of this PR